.github/workflows/doctest.yml

```diff
@@ -0,0 +1,18 @@
+name: doctest
+
+on:
+  pull_request:
+    branches:
+      - master
+
+jobs:
+  doctest:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+
+      - run: cabal update
+
+      # NOTE: We can't use `cabal doctest` here, as `cabal doctest` uses
+      # `--build-depends=QuickCheck`, which results in a dependency cycle.
+      - run: cabal install doctest --ignore-project --overwrite-policy=always && cabal build && cabal repl --build-depends=unliftio --with-compiler=doctest --repl-options='-w -Wdefault'
```

Comment:
@sol This approach will only run doctests for the latest ghc and latest cabal that is provided on the ubuntu image. Would it be possible, instead of adding a whole new workflow, to add this line to the cabal section of the GihubActions workflow here: https://github.com/haskell/random/blob/623cf518c6bc4c1de6a4d59b43c3ae37fcb22a35/.github/workflows/ci.yaml#L87

Reply:
> @sol This approach will only run doctests for the latest ghc and latest cabal that is provided on the ubuntu image.

Which is basically, with very short delays, the latest released version of ghc and cabal.

> Would it be possible, instead of adding a whole new workflow, to add this line to the cabal section of the GihubActions workflow

That makes sense 👍

I don't have the bandwidth to do this myself + I do think this PR is an improvement over the status quo. You may choose to merge this as-is, then somebody can improve on it in a follow up PR. Alternatively, anybody is welcome to take this PR as a template or starting point for a new PR.

I am happy to review any doctest related changes so please always feel free to @-mention me.  I just don't have the capacity to fix everybody's doctests myself.

The important points are:
- Don't use `:set`, use `:seti` instead.
- Don't use `doctest` as a library. Use `cabal repl` or `cabal doctest`. https://github.com/sol/doctest?tab=readme-ov-file#running-doctest-for-a-cabal-package

Reply:
Yet an other option, I granted "maintainer access" to my branch. So I think technically a maintainer could add new commits to my branch (or amend existing commits).

Reply:
No worries, @sol, I'll take it from here. You've been tremendously helpful with this PR already. So, thank you very much!

> I am happy to review any doctest related changes so please always feel free to @-mention me. 

Thank you very much, I'll definitely take you up on that offer! :slightly_smiling_face: